packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -1698,15 +1696,38 @@ function commitRootImpl(root) {
       currentTime,
       remainingExpirationTime,
     );
+
+    if (enableSchedulerTracing) {
+      // Temporarily restore interactions in the case of e.g. hidden subtrees and suspended hydration.
+      // Don't restore otherwise or it would blur interrupting high-pri udpates with low-pri work.
+      if (reschedulePendingInteractionsAtNeverPriority) {
+        scheduleInteractions(root, Never, root.memoizedInteractions);
+      }
+    }
+
     scheduleCallbackForRoot(root, priorityLevel, remainingExpirationTime);
   } else {
     // If there's no remaining work, we can clear the set of already failed
     // error boundaries.
     legacyErrorBoundariesThatAlreadyFailed = null;
   }
 
+  if (enableSchedulerTracing) {
+    if (!rootDidHavePassiveEffects) {
+      // If there are no passive effects, then we can complete the pending interactions.
+      // Otherwise, we'll wait until after the passive effects are flushed.
+      // Wait to do this until after remaining work has been scheduled,
+      // so that we don't prematurely signal complete for interactions when there's e.g. hidden work.
+      finishPendingInteractions(root, expirationTime);
+    }
+  }
+
   onCommitRoot(finishedWork.stateNode, expirationTime);
 
+  if (enableSchedulerTracing) {
+    reschedulePendingInteractionsAtNeverPriority = false;
```

Comment:
Feels fishy that this is in a separate branch from the one above where you check if it's true

Reply:
This is in a broader branch. Seemed best to always unset it, but maybe I should just do that in `prepareFreshStack`